kratos/includes/accessor.h

```diff
@@ -0,0 +1,245 @@
+     * @return The bool type properties
+     */
+    virtual bool GetValue(
+        const Variable<bool>& rVariable,
+        const Properties& rProperties,
+        const GeometryType& rGeometry,
+        const Vector& rShapeFunctionVector,
+        const ProcessInfo& rProcessInfo
+        ) const;
+
+    /**
+     * @brief Custom method to retrieve int type properties
+     * @param rVariable The variable considered (int type properties)
+     * @param rProperties The properties considered
+     * @param rGeometry The geometry considered
+     * @param rShapeFunctionVector The shape function of the GP considered
+     * @param rProcessInfo The process info considered
+     * @return The int type properties
+     */
+    virtual int GetValue(
+        const Variable<int>& rVariable,
+        const Properties& rProperties,
+        const GeometryType& rGeometry,
+        const Vector& rShapeFunctionVector,
+        const ProcessInfo& rProcessInfo
+        ) const;
+
+    /**
+     * @brief Custom method to retrieve Matrix type properties
+     * @param rVariable The variable considered (Matrix type properties)
+     * @param rProperties The properties considered
+     * @param rGeometry The geometry considered
+     * @param rShapeFunctionVector The shape function of the GP considered
+     * @param rProcessInfo The process info considered
+     * @return The Matrix type properties
+     */
+    virtual Matrix GetValue(
+        const Variable<Matrix>& rVariable,
+        const Properties& rProperties,
+        const GeometryType& rGeometry,
+        const Vector& rShapeFunctionVector,
+        const ProcessInfo& rProcessInfo
+        ) const;
+
+    /**
+     * @brief Custom method to retrieve array_1d<double, 3 > type properties
+     * @param rVariable The variable considered (array_1d<double, 3 > type properties)
+     * @param rProperties The properties considered
+     * @param rGeometry The geometry considered
+     * @param rShapeFunctionVector The shape function of the GP considered
+     * @param rProcessInfo The process info considered
+     * @return The array_1d<double, 3 > type properties
+     */
+    virtual array_1d<double, 3> GetValue(
+        const Variable<array_1d<double, 3>>& rVariable,
+        const Properties& rProperties,
+        const GeometryType& rGeometry,
+        const Vector& rShapeFunctionVector,
+        const ProcessInfo& rProcessInfo
+        ) const;
+
+    /**
+     * @brief Custom method to retrieve array_1d<double, 6 > type properties
+     * @param rVariable The variable considered (array_1d<double, 6 > type properties)
+     * @param rProperties The properties considered
+     * @param rGeometry The geometry considered
+     * @param rShapeFunctionVector The shape function of the GP considered
+     * @param rProcessInfo The process info considered
+     * @return The array_1d<double, 6 > type properties
+     */
+    virtual array_1d<double, 6> GetValue(
```

Comment:
Just wondering, we have `array_1d<double, 3>`, `array_1d<double, 4>`, `array_1d<double, 6>`, `array_1d<double, 9>` interfaces in core. Is this intentional to limit it to these types only?

Reply:
I will add them! :)

Reply:
Thanks :)